skbuild/constants.py

```diff
@@ -42,14 +42,17 @@ def _default_skbuild_plat_name():
     # Use CMAKE_OSX_ARCHITECTURES if that is set, otherwise use ARCHFLAGS,
     # which is the variable used by Setuptools. Fall back to the machine arch
     # if neither of those is given.
-    machine = os.environ.get("ARCHFLAGS", machine)
+    archflags = os.environ.get("ARCHFLAGS")
+    if archflags is not None:
+        machine = ";".join(set(archflags.split()) & {"x86_64", "arm64"})
```

Comment:
may be sensible to introduce an intermediate variable `supported_macos_architectures={"x86_64", "arm64"}`  ?

Reply:
Sure, probably `SUPPORTED_MACOS_ARCHETECTURES` if it's a constant. I was tempted to do this originally, but it would not have simplified the universal line below, because that is uniquely these two archs, not all supported archs (which currently is just the same two, of course).

Reply:
Since these are only used in this function ... may be this would make things more explicit ?


```diff
    # Use CMAKE_OSX_ARCHITECTURES if that is set, otherwise use ARCHFLAGS,
    # which is the variable used by Setuptools. Fall back to the machine arch
    # if neither of those is given.
    archflags = os.environ.get("ARCHFLAGS")
    if archflags is not None:
+       supported_macos_architectures = {"x86_64", "arm64"}
-        machine = ";".join(set(archflags.split()) & {"x86_64", "arm64"})
+        machine = ";".join(set(archflags.split()) & supported_macos_architectures)

    machine = os.environ.get("CMAKE_OSX_ARCHITECTURES", machine)

    # Handle universal2 wheels, if those two architectures are requested.
+    macos_universal2_architectures = macos_architectures
-    if set(machine.split(";")) == {"x86_64", "arm64"}:
+    if set(machine.split(";")) == macos_universal_architectures:
        machine = "universal2"
```

May also be sensible to update the docstring of `_default_skbuild_plat_name`

Reply:
That looks good. Not sure I'd do `macos_universal2_architectures = macos_architectures` over `macos_universal2_architectures = {"x86_64", "arm64"}` - if you were to add something to `macos_architectures`, it would not be added to `macos_universal2_architectures`. Keeping code DRY is good, but not if the abstraction is wrong - [AHA](https://testandcode.com/160).

Reply:
Thanks for the AHA reference, I will read it through :pray: 

What about ?

```diff
    # Use CMAKE_OSX_ARCHITECTURES if that is set, otherwise use ARCHFLAGS,
    # which is the variable used by Setuptools. Fall back to the machine arch
    # if neither of those is given.
+   supported_macos_architectures = {"x86_64", "arm64"}
    archflags = os.environ.get("ARCHFLAGS")
    if archflags is not None:
-        machine = ";".join(set(archflags.split()) & {"x86_64", "arm64"})
+        machine = ";".join(set(archflags.split()) & supported_macos_architectures)

    machine = os.environ.get("CMAKE_OSX_ARCHITECTURES", machine)

    # Handle universal2 wheels, if those two architectures are requested.
-    if set(machine.split(";")) == {"x86_64", "arm64"}:
+    if set(machine.split(";")) == supported_macos_architectures:
        machine = "universal2"
```

Reply:
I would do:

```diff
    # Use CMAKE_OSX_ARCHITECTURES if that is set, otherwise use ARCHFLAGS,
    # which is the variable used by Setuptools. Fall back to the machine arch
    # if neither of those is given.
+   supported_macos_architectures = {"x86_64", "arm64"}
    archflags = os.environ.get("ARCHFLAGS")
    if archflags is not None:
-        machine = ";".join(set(archflags.split()) & {"x86_64", "arm64"})
+        machine = ";".join(set(archflags.split()) & supported_macos_architectures)

    machine = os.environ.get("CMAKE_OSX_ARCHITECTURES", machine)

    # Handle universal2 wheels, if those two architectures are requested.
+   macos_universal2_architectures = {"x86_64", "arm64"}
-   if set(machine.split(";")) == macos_universal2_architectures:
        machine = "universal2"
```

Or,  maybe, you could set `supported_macos_architectures = macos_universal2_architectures` - this direction is correct. If Apple switches to a new architecture some day, it is clear you need to now add one to `macos_universal2_architectures` to get `supported_macos_architectures`.